rust/kernel/spi.rs

```diff
@@ -0,0 +1,279 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! This module provides safer and higher level abstraction over the kernel's SPI types
+//! and functions.
+//!
+//! C header: [`include/linux/spi/spi.h`](../../../../include/linux/spi/spi.h)
+
+use crate::bindings;
+use crate::c_types;
+use crate::error::{Error, Result};
+use crate::str::CStr;
+use alloc::boxed::Box;
+use core::pin::Pin;
+
+/// Wrapper struct around the kernel's `spi_device`
+#[derive(Clone, Copy)]
+pub struct SpiDevice(*mut bindings::spi_device);
+
+impl SpiDevice {
+    /// Create an SpiDevice from a mutable spi_device raw pointer. This function is unsafe
+    /// as the pointer might be invalid.
```

Comment:
The safety comment is not needed since we have the `# Safety` section (and anyway one cannot call `unsafe` functions without a block, so that is fine).

Also please add intra-doc links wherever possible and use "code" marks too, e.g.:

```suggestion
    /// Create an [`SpiDevice`] from a mutable `spi_device` raw pointer.
```

Reply:
Fixed by 92a19c2ef2d0d0780c34327b273a21db627af711